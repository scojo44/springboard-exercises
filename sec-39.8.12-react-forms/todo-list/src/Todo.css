.Todo {
  list-style: none;
}
.Todo button {
  margin-left: 1rem;
}