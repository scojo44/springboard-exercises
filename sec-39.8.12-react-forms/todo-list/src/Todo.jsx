import React, { useState } from 'react'
import Task from './Task'
import TaskEdit from './TaskEdit'
import './Todo.css'

function Todo({id, task, remove, update, completed = false, editing = false}) {
  return (
    <li className='Todo'>
      {editing
      ? <TaskEdit id={id} task={task} update={update} completed={completed} />
      : <Task id={id} task={task} remove={remove} update={update} completed={completed} />
      }
    </li>
  )
}

export default Todo
