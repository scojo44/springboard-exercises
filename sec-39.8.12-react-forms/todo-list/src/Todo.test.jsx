import { render, fireEvent } from '@testing-library/react';
import Todo from './Todo';

describe('Todo Tests', () => {
  it('renders without crashing', () => {
    render(<Todo id={1} task="Learn React" />);
  });

  it('matches snapshot', () => {
    const {asFragment} = render(<Todo id={1} task="Learn React" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('shows a task', () => {
    const {getByText} = render(<Todo id={1} task="Learn React" remove={x => x} update={x => x} />);
    const task = getByText('Learn React');
    expect(task).toBeInTheDocument();
  });

  it('shows a task in editing mode', () => {
    const {getByText} = render(<Todo id={1} task="Learn React" editing={true} />);
    const button = getByText('Save');

    expect(button).toBeInTheDocument();
  });
});
