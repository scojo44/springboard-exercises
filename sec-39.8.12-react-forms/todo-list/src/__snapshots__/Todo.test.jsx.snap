// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Todo Tests > matches snapshot 1`] = `
<DocumentFragment>
  <li
    class="Todo"
  >
    <div
      class="Task"
    >
      <input
        id="toggle-1"
        type="checkbox"
      />
      <label
        class=""
        for="toggle-1"
      >
        Learn React
      </label>
    </div>
    <div>
      <button>
        Edit
      </button>
    </div>
    <div>
      <button>
        X
      </button>
    </div>
  </li>
</DocumentFragment>
`;
